nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/PayaraHazelcastTenant.java

```
@@ -0,0 +1,218 @@
+        }
+    }
+
+    @Override
+    public void registerObject(DestroyEventContext destroyContext) {
+        destroyEventListener = new EventListenerImpl(destroyContext);
+        events.register(destroyEventListener);
+    }
+
+    @Override
+    public void unregisterObject() {
+        // Hazelcast object has been destroyed
+        events.unregister(destroyEventListener);
+        destroyEventListener = null;
+    }
+
+    @Override
+    public Closeable setTenant() {
+        try {
+            return contextInstance.pushRequestContext()::close;
+        } catch (IllegalStateException exc) {
+            throw exc;
+        }
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeObject(contextInstance);
+        out.writeUTF(moduleName);
+    }
+
+    @Override
+    public void readData(ObjectDataInput in) throws IOException {
+        contextInstance = in.readObject();
+        moduleName = in.readUTF();
+    }
+
+    @Override
+    public boolean isAvailable(Tenantable tenantable) {
+        if (!contextInstance.isLoaded()) {
+            if (!tenantable.requiresTenantContext() || tenantNotRequired(tenantable)) {
+                return true;
+            }
+            lock.lock();
+            try {
+                String componentId = contextInstance.getInstanceComponentId();
+                int unavailableCount = blockedCounts.compute(componentId, (k, v) -> v == null ? 0 : ++v);
+                log.log(unavailableCount > 100 ? Level.INFO : Level.FINEST,
+                        String.format("BLOCKED: tenant not available: %s, module %s, Operation: %s",
+                                componentId, moduleName, tenantable.getClass().getName()));
+                if (unavailableCount > 100) {
+                    blockedCounts.remove(componentId);
+                }
+                condition.await(100, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException ex) {
```

are you intentionally swallowing the exception?